app/controllers/helpers/DeclarationHolderHelper.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.helpers
+
+import forms.declaration.declarationHolder.DeclarationHolder
+import models.requests.JourneyRequest
+import play.api.data.{Form, FormError}
+
+object DeclarationHolderHelper {
+
+  val DeclarationHolderFormGroupId: String = "declarationHolder"
+
+  def declarationHolders(implicit request: JourneyRequest[_]): Seq[DeclarationHolder] = request.cacheModel.declarationHolders
+
+  def form(implicit request: JourneyRequest[_]): Form[DeclarationHolder] =
+    DeclarationHolder.form(request.eori, request.cacheModel.additionalDeclarationType)
+
+  private val mutuallyExclusiveAuthorisationCodes = List("CSE", "EXRR")
+
+  def validateMutuallyExclusiveAuthCodes(maybeHolder: Option[DeclarationHolder], holders: Seq[DeclarationHolder]): Option[FormError] =
+    maybeHolder match {
+      case Some(DeclarationHolder(Some(code), _, _)) if mutuallyExclusiveAuthorisationCodes.contains(code) =>
+        val mustNotAlreadyContainCodes = mutuallyExclusiveAuthorisationCodes.filter(_ != code)
+
+        if (!holders.map(_.authorisationTypeCode.getOrElse("")).containsSlice(mustNotAlreadyContainCodes)) None
+        else Some(FormError(DeclarationHolderFormGroupId, s"declaration.declarationHolder.${code}.error.exclusive"))
+
+      case _ => None
+    }
```

Comment:
Could this method be moved into the `DeclarationHolder` companion object? Then at least all the validation logic would be in one place :slightly_smiling_face:

Reply:
Actually no. The CSE-EXRR validation must only take place when adding an authorisation, and not when changing one. The `form` can only include validations supporting both pages (add and change).

Liz's words...
> I think that I would allow this to happen. The user will get a DMSREJ. We cannot protect them from every possible thing they want to do wrong

Reply:
Can't see a unit test for this method, think it would be useful to test this is isolation (rather than relying on the controller's unit tests)